Library/Homebrew/cmd/list.rb

```diff
@@ -86,7 +86,13 @@ def list
         ls_args << "-r" if args.r?
         ls_args << "-t" if args.t?
 
-        safe_system "ls", *ls_args, HOMEBREW_CELLAR
+        if !$stdout.tty? && !args.formula?
+          odeprecated "`brew list` to only list formulae", "`brew list --formula`"
+          safe_system "ls", *ls_args, HOMEBREW_CELLAR
+        else
+          safe_system "ls", *ls_args, HOMEBREW_CELLAR
+          print list_casks(args: args) unless args.formula?
```

Comment:
```suggestion
         list_casks(args: args) unless args.formula?
```

Reply:
Hm, it throws error because it return list_casks and not list Cellar, right?